pkg/network/ebpf/c/skb.h

```diff
@@ -52,11 +56,9 @@ static __always_inline int sk_buff_to_tuple(struct sk_buff *skb, conn_tuple_t *t
                 return 0;
         }
         trans_len = iph.tot_len - (iph.ihl * 4);
-        bpf_probe_read_kernel_with_telemetry(&tup->saddr_l, sizeof(__be32), &iph.saddr);
-        bpf_probe_read_kernel_with_telemetry(&tup->daddr_l, sizeof(__be32), &iph.daddr);
-    }
-#ifdef FEATURE_IPV6_ENABLED
-    else if (iph.version == 6) {
+        BPF_CORE_READ_INTO((__be32*)(&tup->saddr_l), &iph, saddr);
+        BPF_CORE_READ_INTO((__be32*)(&tup->daddr_l), &iph, daddr);
```

Comment:
I don't think you need the `&` since you are using the macro, and commas separate pointer indirection not direct field access.
```suggestion
        BPF_CORE_READ_INTO((__be32*)(&tup->saddr_l), iph.saddr);
        BPF_CORE_READ_INTO((__be32*)(&tup->daddr_l), iph.daddr);
```

Reply:
That's what I did earlier, but `BPF_CORE_READ_INTO` take three arguments minimum.

Reply:
tbh, I don't think you need a read at all here. The layout of `iphdr` is not going to change, and you already have it on the stack. So you can probably get away with:
`tup->saddr_l = iph.saddr`
You may need to add some casting, as long as it doesn't sign-extend.

Reply:
What you have now is definitely wrong though. It is equivalent to `&&iph->saddr`.